src/types.ts

```diff
@@ -77,3 +77,130 @@ export interface PaymentEligibilityPayload extends PaymentPayloadBase {
 export interface PaymentPayload extends PaymentPayloadBase {
   payment: AtLeastOne<PaymentDataProps, 'billing_address' | 'shipping_address'>
 }
+
+
+export type PaymentId = string
+export interface Card {
+  brand: string
+  country: string
+  created: number
+  exp_month: number
+  exp_year: number
+  funding: string
+  id: string
+  last4: string
+  psp_representations: {
+    stripe: {
+      customer_id: string
+      original_source_id: string
+    }
+  }
+  three_d_secure_possible: boolean
+  verified: boolean
+}
+export interface Installment {
+  customer_fee: number
+  due_date: number
+  original_purchase_amount: number
+  purchase_amount: number
+  // TODO complete
+  state: 'pending'
+}
+export interface Requirement {
+  is_met: boolean
+  // TODO complete
+  name:
+    | 'customer_info'
+    | 'phone_verification'
+    | 'id_video_verification'
+    | 'banking_data_verification'
+}
+export type State =
+  | 'not_ready'
+  | 'not_started'
+  | 'scored_no'
+  | 'scored_maybe'
+  | 'scored_yes'
+  | 'in_progress'
+  | 'paid'
+  | 'late'
+  | 'default'
+
+export interface Payment {
+  billing_address: null
```

Comment:
Shouldn't we have a `Address` type and use that here and with `shipping_address` ? ^^